### PR TITLE
lates version of perl fails with the example code, need to use v5.34.0

### PR DESCRIPTION
**Pull request by mhus**
I tried the example and it is no more working. Perl will not process pi with an error. Perl until v5.34.0 will support the command. This fix will do for now but you should use another example working with the actual version of perl.

### CODE REVIEW

**Comment by mhus** [comment]
Found an issue for this topic #34073

**Comment by JohnTitor** [comment]
I've already opened #34070 to avoid the root issue.

**Comment by mhus** [comment]
Hi, I can't get the 'lgtm' label. Don't know why... any ideas?

**Comment by jihoon-seo** [comment]
/lgtm

Refs:
- #34073
- #34475
- #34070

**Comment by jihoon-seo** [comment]
@mhus Could you replace `image: perl:5.34` with `image: perl:5.34.0` in `content/en/examples/controllers/job.yaml` as well,
in this PR?
(To do this, you might need to rebase with the latest `main`.)

It's for consistency and for pinning the container image version more specifically.
(Currently `5.34` equals to `5.34.1`, 
so following the steps in https://kubernetes.io/docs/concepts/workloads/controllers/job/ will lead to pull two different versions of the `perl` container image.)

/lgtm cancel

**Comment by kbhawkey** [comment]
Preview:
https://deploy-preview-34414--kubernetes-io-main-staging.netlify.app/docs/concepts/workloads/controllers/job/#running-an-example-job

**Comment by kbhawkey** [comment]
@mhus , The changes look good.
If you have time, could you also update the example file (`content/en/examples/controllers/job.yaml`) to define the image as `perl:5.34.0`? The example file currently uses `perl:5.34`.
Thanks.

**Comment by mhus** [comment]
Something in the preview is wrong. In my pull request is in line (it's not an include)

43 kubectl apply -f https://kubernetes.io/examples/controllers/job.yaml

In the preview is shown the full source code from job.yaml - and it not includes the perl version.

File: content/en/examples/controllers/job.yaml [hunk 1]
```diff
@@ -7,8 +7,8 @@ spec:
     spec:
       containers:
       - name: pi
-        image: perl:5.34
-        command: ["perl",  "-Mbignum=bpi", "-wle", "'print bpi(2000)'"]
+        image: perl:5.34.0
+        command: ["perl",  "-Mbignum=bpi", "-wle", "print bpi(2000)"]
```
Comment (kbhawkey) [inline content/en/examples/controllers/job.yaml:11]: Hi @mhus . Do you need single quotes around the `print` command (I'm not sure why the quotes are needed)? If not, would you remove the single quotes?
The other changes look good to me. Thank you!

**Review by mhus — COMMENTED** [review]
Looks final now.

**Review by soltysh — APPROVED** [review]
/lgtm

**Pull request merged**